### PR TITLE
Fix segfault when baking without a config

### LINKED ISSUE #388: PyOpenColorIO Baker segfault
```
>>> import PyOpenColorIO
>>> PyOpenColorIO.version
'1.0.9'
>>> b = PyOpenColorIO.Baker() 
>>> b.setFormat("cinespace")
>>> b.bake()
Segmentation fault
```

The specific format isn't important, happens with "lustre", "hoduini" etc - but if unset it gives a useful `PyOpenColorIO.Exception: The format named '' could not be found.` error

### PR DESCRIPTION
**Pull request by matthewdcong**
Fixes #388. The file format implementations call:
`ConstConfigRcPtr config = baker.getConfig()` followed by `config->getProcessor`. However, if a configuration has not been set for the Baker, config will be a null `shared_ptr` and the `getProcessor` call with segfault. Added a unit test to verify that an exception is properly thrown and verified that the Python script in the issue no longer segfaults.

### CODE REVIEW

File: src/OpenColorIO/Baker.cpp [hunk 1]
```diff
@@ -213,7 +213,14 @@ OCIO_NAMESPACE_ENTER
             err << "' could not be found. ";
             throw Exception(err.str().c_str());
         }
-        
+
+        if(!getConfig())
+        {
+            std::ostringstream err;
+            err << "No OCIO config has been set";
+            throw Exception(err.str().c_str());
```
Comment (hodoulp) [inline src/OpenColorIO/Baker.cpp]: These three lines could be merged in a single line like:
`throw Exception("No OCIO config has been set.");`

File: src/OpenColorIO/Baker.cpp [hunk 2]
```diff
@@ -388,6 +395,16 @@ OCIO_ADD_TEST(Baker_Unit_Tests, bake)
 
 }
 
+OCIO_ADD_TEST(Baker_Unit_Tests, bake_empty_config)
```
Comment (hodoulp) [inline src/OpenColorIO/Baker.cpp]: [Minor] To better follow other unit test names, the 'major' name is the class name i.e. 'Baker', and the 'minor' name is the test itself i.e. 'empty_config' or 'missing_config'.

File: src/OpenColorIO/Baker.cpp [hunk 3]
```diff
@@ -388,6 +395,16 @@ OCIO_ADD_TEST(Baker_Unit_Tests, bake)
...
+{
+    // Verify that running bake with an empty configuration
+    // throws an exception and does not segfault.
+    OCIO::BakerRcPtr bake = OCIO::Baker::Create();
+    bake->setFormat("cinespace");
```
Comment (hodoulp) [inline src/OpenColorIO/Baker.cpp]: [Request] The `setFormat()` method does not validate that the format accepts the 'bake' mode.  That could be something interesting to add if you have time.

**Comment by hodoulp** [comment]
**Good candidate for OCIOv1.**

File: src/OpenColorIO/Baker.cpp (hunk 1 above)
Comment (matthewdcong) [reply src/OpenColorIO/Baker.cpp]: Merged.

File: src/OpenColorIO/Baker.cpp (hunk 2 above)
Comment (matthewdcong) [reply src/OpenColorIO/Baker.cpp]: Got it, thanks! Changed the major name to `Baker` for both the `bake` and `empty_config` tests.

File: src/OpenColorIO/Baker.cpp (hunk 3 above)
Comment (matthewdcong) [reply src/OpenColorIO/Baker.cpp]: Added the validation.

**Comment by matthewdcong** [comment]
Thanks for reviewing, Patrick! I've made the changes suggested and would be happy to backport this to OCIOv1 in the near future.

**Pull request merged**